dangerzone/isolation_provider/container.py

```diff
@@ -304,11 +305,33 @@ def _convert_with_tmpdirs(
                 f"Conversion output (doc to pixels):\n{self.sanitize_conversion_str(untrusted_log)}"
             )
 
+        num_pages = len(glob.glob(f"{pixel_dir}/page-*.rgb"))
+        self.verify_received_pixel_files(pixel_dir, num_pages)
+
+        for page in range(1, num_pages + 1):
+            filename_base = f"{pixel_dir}/page-{page}"
+            rgb_filename = f"{filename_base}.rgb"
+            width_filename = f"{filename_base}.width"
+            height_filename = f"{filename_base}.height"
+            with open(width_filename) as f:
+                width = int(f.read().strip())
+            with open(height_filename) as f:
+                height = int(f.read().strip())
+            with open(rgb_filename, "rb") as rgb_f:
+                untrusted_pixels = rgb_f.read()
+                self.convert_pixels_to_png(
+                    str(pixel_dir), page, width, height, rgb_data=untrusted_pixels
+                )
+
+            os.remove(rgb_filename)
+            os.remove(width_filename)
+            os.remove(height_filename)
```

Comment:
All of this should be move to after checking if there was a non-zero exit code.

Reply:
Just to make sure why this code is necessary in the first place:

1. The **current** containers version, which does not stream the data to the host yet, has to write all the pixels in the temporary directory. This is the worst case scenario, and by converting them to PNG after the fact does not help much.
2. The **streaming** (post-#443) containers version will piggyback on the Qubes code that converts the RGB images to PNG on the fly, which has already been added in this PR.

Still, the code seems to be necessary because it makes `pixels-to-pdf` accept only PNG images, not RGB, thereby simplifying its interface. Is this your rationale as well?

Reply:
> Still, the code seems to be necessary because it makes pixels-to-pdf accept only PNG images, not RGB, thereby simplifying its interface. Is this your rationale as well?

Yes, that's the logic here. And it's not worth leaving the pixels-to-pdf code more complex to make #412 easier.